apps/server/src/modules/legacy-school/repo/school-system-options.repo.ts

```diff
@@ -0,0 +1,94 @@
+import { EntityManager } from '@mikro-orm/mongodb';
+import { Injectable } from '@nestjs/common';
+import { SystemEntity } from '@shared/domain/entity';
+import { SystemProvisioningStrategy } from '@shared/domain/interface/system-provisioning.strategy';
+import { EntityId } from '@shared/domain/types';
+import {
+	AnyProvisioningOptions,
+	SchoolSystemOptions,
+	SchoolSystemOptionsBuilder,
+	SchoolSystemOptionsProps,
+} from '../domain';
+import { SchoolSystemOptionsEntity, SchoolSystemOptionsEntityProps } from '../entity';
+import { MissingProvisioningStrategyLoggableException } from '../loggable';
+import { SchoolSystemOptionsRepoMapper } from './school-system-options-repo.mapper';
+
+@Injectable()
+export class SchoolSystemOptionsRepo {
+	constructor(private readonly em: EntityManager) {}
+
+	public async findBySchoolIdAndSystemId(schoolId: EntityId, systemId: EntityId): Promise<SchoolSystemOptions | null> {
+		const entity: SchoolSystemOptionsEntity | null = await this.em.findOne(
+			SchoolSystemOptionsEntity,
+			{
+				school: schoolId,
+				system: systemId,
+			},
+			{ populate: ['system.provisioningStrategy'] }
+		);
+
+		if (!entity) {
+			return null;
+		}
+
+		if (!entity.system.provisioningStrategy) {
+			throw new MissingProvisioningStrategyLoggableException(entity.system.id);
+		}
+
+		const domainObject: SchoolSystemOptions = this.buildDomainObject(entity, entity.system.provisioningStrategy);
+
+		return domainObject;
+	}
+
+	public async save(domainObject: SchoolSystemOptions): Promise<SchoolSystemOptions> {
+		const entityProps: SchoolSystemOptionsEntityProps = SchoolSystemOptionsRepoMapper.mapDomainObjectToEntityProperties(
+			domainObject,
+			this.em
+		);
+
+		const newEntity: SchoolSystemOptionsEntity = new SchoolSystemOptionsEntity(entityProps);
+
+		const existingEntity: SchoolSystemOptionsEntity | null = await this.em.findOne(SchoolSystemOptionsEntity, {
+			id: domainObject.id,
+		});
+
+		const system: SystemEntity | null = await this.em.findOne(SystemEntity, {
+			id: domainObject.systemId,
+		});
+
+		if (!system?.provisioningStrategy) {
+			throw new MissingProvisioningStrategyLoggableException(domainObject.systemId);
+		}
+
+		let savedEntity: SchoolSystemOptionsEntity;
+		if (existingEntity) {
+			savedEntity = this.em.assign(existingEntity, newEntity);
```

Comment:
From the MikroORM Dokumentation:
> MikroORM uses the Identity Map pattern to track objects. Whenever you fetch an object from the database, MikroORM will keep a reference to this object inside its UnitOfWork. This allows MikroORM room for optimizations. If you call the EntityManager and ask for an entity with a specific ID twice, it will return the same instance.
```
const jon1 = await authorRepository.findOne(1);
const jon2 = await authorRepository.findOne(1);
```
>Only one SELECT query will be fired against the database here. In the second findOne() call MikroORM will check the identity map first and will skip the database round trip as it will find the entity already loaded.